presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -729,6 +736,65 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
                 new PartitionStatistics(basicStatistics, ImmutableMap.of()));
     }
 
+    @Override
+    public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<ColumnMetadata> columns, Optional<ConnectorPartitioningMetadata> partitioning)
+    {
+        String schemaName = getTemporaryTableSchema(session);
+        String tableName = "__presto_temporary_table_" + randomUUID().toString().replaceAll("-", "_");
```

Comment:
Should this begin with a `.`?

Reply:
Why?

But anyhow, i don't think that's a good idea. The LocationService implementation may not support special symbols in the table name (e.g.: our internal LocationService only supports alphanumerics + `_`)